lib/chef/knife/bmcs_server_show.rb

```diff
@@ -38,38 +39,28 @@ class BmcsServerShow < Knife
              description: 'The OCID of the server to display. (required)'
 
       def lookup_compartment_name(compartment_id)
-        compartments = identity_client.list_compartments(bmcs_config.tenancy, {})
-        compartments.data && compartments.data.each do |compartment|
-          next unless compartment.id == compartment_id
-          return compartment.description || compartment.name
-        end
+        compartment = identity_client.get_compartment(compartment_id, {})
+        compartment.data.description || compartment.data.name
```

Comment:
I would just go with the name as the primary identifier.

Reply:
This has been done, but the example in the issue is more representative of the 'description' field, not the name.  Which is why I had it using the description first, and if unset, then the name.